askbot/views/users.py

```diff
@@ -1487,12 +1487,45 @@ def user(request, id, tab_name=None):
         user_logged_in=profile_owner.is_authenticated,
     )
 
+    question_filter = {}
+    if request.user != user:
+        question_filter['is_anonymous'] = False
+
+    if askbot_settings.CONTENT_MODERATION_MODE == 'premoderation':
+        question_filter['approved'] = True
+
+    #
+    # Questions
+    #
+    questions_qs = profile_owner.posts.get_questions(
+                    user=request.user
+                ).filter(
+                    **question_filter
+                ).order_by(
+                    '-points'#, '-thread__last_activity_at' to match sorting with ajax loads
+                ).select_related(
+                    'thread', 'thread__last_activity_by'
+                )
+
+    q_paginator = Paginator(questions_qs, const.USER_POSTS_PAGE_SIZE)
```

Comment:
I wonder if we can cut these lines about the paginator out here, since all we need is the count.

I bet we can shorten the calculation of "how many total questions" to this, above at 1500:
```
questions_count = profile_owner.posts.get_questions(
                    user=request.user
                ).filter(
                    **question_filter
                ).count()
```